src/architecture/serialization.md

```diff
@@ -243,6 +243,33 @@ in the correspondence with [the validation rules](#segment-validation-rules).
     binary format all fields of such structure are placed in the header, its body
     is empty. So such a structure is fixed-length.
 
+#### Structure composition
+
+`encoding_struct!` macro allows to make a composition of existing structures:
+
+```Rust
+encoding_struct! {
+  struct Pair {
+    const SIZE = 8;
+    field first: u32 [00 => 04]
+    field second: u32 [04 => 08]
+  }
+}
+
+encoding_struct! {
+  struct Pairs{
+    const SIZE = 8;
+    field inner : Vec<Pair> [00 => 08]
+  }
+}
+```
+
+Here `Pair` is used as a base type in `Pairs` declaration.
+
+!!! note
+    `message!` macro may be used instead of `encoding_struct!` for the outer
+    structure.
```

Comment:
Do you mean this is the case when the outer "structure" is actually a message? (That is, a message can incorporate structure fields in its body.) If this is the case, maybe describe it like:

> Analogously, fields derived from structures defined with `encoding_struct!` macro can be used within messages.

I'd also add an explicit example just to be clear.